Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Generator.cs

```diff
@@ -80,13 +84,29 @@ public void Execute(GeneratorExecutionContext context)
                     }
                 }
 
+                var isExecutable = false;
+                
+                var assemblyAttributes = context.Compilation.Assembly.GetAttributes();
+                
+                // Let's find the AssemblyTitleAttribute
+                var generateMainAttribute = assemblyAttributes
+                    .FirstOrDefault(attr => attr.AttributeClass.Name == nameof(LambdaGenerateMainAttribute));
+
+                if (generateMainAttribute != null)
+                {
+                    isExecutable = true;
```

Comment:
Check the `context.Compilation.CommonOptions.OutputKind` property to see if it is [OutputKind.ConsoleApplication](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.outputkind?view=roslyn-dotnet-4.6.0). Then write a diagnostic error. I forget to change this when I was converting the blueprint in Visual Studio to use this feature and had to figure out after deployment why I was getting an `Internal Error` which is not a great experience.